libgalois/src/analytics/subgraph_extraction/subgraph_extraction.cpp

```diff
@@ -29,6 +29,45 @@ namespace {
 using namespace katana::analytics;
 using edge_iterator = boost::counting_iterator<uint64_t>;
 
+/**
+ * Like std::lower_bound but doesn't dereference iterators. Returns the first
+ * element for which comp is not true.
+ */
+template <typename Iterator, typename Compare>
+Iterator
+LowerBound(Iterator first, Iterator last, Compare comp) {
+  using difference_type =
+      typename std::iterator_traits<Iterator>::difference_type;
+
+  Iterator it;
+  difference_type count;
+  difference_type half;
+
+  count = std::distance(first, last);
+  while (count > 0) {
+    it = first;
+    half = count / 2;
+    std::advance(it, half);
+    if (comp(it)) {
+      first = ++it;
+      count -= half + 1;
+    } else {
+      count = half;
+    }
+  }
+  return first;
+}
+
+template <typename G>
+struct LessThan {
+  const G& g;
+  typename G::Node n;
+  LessThan(const G& g, typename G::Node n) : g(g), n(n) {}
+  bool operator()(typename G::edge_iterator it) {
+    return *g.GetEdgeDest(it) < n;
+  }
+};
+
```

Comment:
This is copied from `triangle_count.cpp`, please lift into a header instead, and use that header in both places. We already have `analytics/Utils.h`, but something in the source directory would be fine too. Duplicated code is a maintainable nightmare.